client/src/reducers/teaReducers.js

```diff
@@ -1,20 +1,23 @@
-import { Set, Map } from "immutable";
-
-export default (state = [], action) => {
+export default (state = { allTeas: {}, teaIDs: [] }, action) => {
   switch (action.type) {
     case "ADD_TEA":
-      return [...state, action.payload];
-    case "DELETE_TEA":
-      return state.filter(t => t.id !== action.payload.id);
     case "EDIT_TEA":
-      return state.map(t => (t.id === action.payload.id ? action.payload : t));
+      let updateAllTeas = {
+        ...state.allTeas,
+        ...{ [action.payload.id]: action.payload }
+      };
+      let updateIDsArr = [...state.teaIDs, action.payload.id];
+
+      return { allTeas: updateAllTeas, teaIDs: updateIDsArr };
+
+    case "DELETE_TEA":
+      let { [action.payload]: val, ...newAllTeas } = state.allTeas;
```

Comment:
IMO, no reason to remove the tea from the `allTeas` list. If its ID doesn't exist in `teaIDs` that should be good enough. Sometimes you might want to keep the object in memory so you can refer to it having been deleted or keep it so you can undo.